src/datasets/builder.py

```diff
@@ -727,8 +729,12 @@ def _save_infos(self):
 
     def _make_split_generators_kwargs(self, prepare_split_kwargs):
         """Get kwargs for `self._split_generators()` from `prepare_split_kwargs`."""
-        del prepare_split_kwargs
-        return {}
+        split_generators_kwargs = {}
+        split_generators_arg_names = inspect.signature(self._split_generators).parameters.keys()
+        if "streaming" in split_generators_arg_names:
+            streaming = isinstance(prepare_split_kwargs.get("dl_manager"), StreamingDownloadManager)
```

Comment:
I guess you need the `DownloadManager` instance to find out whether we are in streaming mode or not... Because the builder itself knows nothing about streaming or not...

Reply:
Indeed having this logic inside the `builder` can be a bit unexpected. An alternative would be to replace the `streaming` parameter by
```python
streaming = dl_manager.is_streaming()
```
inside the dataset script